src/main/java/com/hotels/hcommon/hive/metastore/iterator/PartitionIterator.java

```diff
@@ -31,37 +31,61 @@
 /** Iterates over partitions, lazily loading in batches where possible. */
 public class PartitionIterator implements Iterator<Partition> {
 
+  public enum Ordering {
+    NATURAL,
+    REVERSE
+  }
+
   private static final Logger LOG = LoggerFactory.getLogger(PartitionIterator.class);
 
   private static final short NO_LIMIT = (short) -1;
 
-  private static List<String> loadPartitionNames(IMetaStoreClient metastore, Table table)
+  private static List<String> loadPartitionNames(IMetaStoreClient metastore, Table table, Ordering ordering)
     throws MetaException, TException {
     // partition names: (key=value/)*(key=value)
     LOG.debug("Fetching all partition names.");
     List<String> names = metastore.listPartitionNames(table.getDbName(), table.getTableName(), NO_LIMIT);
+    if (ordering == Ordering.REVERSE) {
```

Comment:
should this be .equals ? I know for ENUMS, it doesn't make any difference. Are you worried that ordering can be null here ?

Reply:
I think `==` is better for enums I think. 
Not really worried about null, someone would have to forcefully call it with null an exception is ok in that case. There are convenience constructors so no existing code has to change